### PR TITLE
HTML: document.open() and document's URL (with fragments)

### PR DESCRIPTION
**Pull request by annevk**
For https://github.com/whatwg/html/pull/3651.

### CODE REVIEW

**Comment by annevk** [comment]
The interesting thing is that if you reload this frame, it'll use the original URL, so that is preserved somehow. What probably happens is that reloading is based in history entries.

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url-fragment.window.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  t.add_cleanup(() => { frame.remove(); });
+  assert_equals(frame.contentDocument.URL, "about:blank");
+  assert_equals(frame.contentWindow.location.href, "about:blank");
+  self.onhashchange = t.step_func_done(() => {
+    frame.contentDocument.open();
+    assert_equals(frame.contentDocument.URL, urlSansHash);
+    assert_equals(frame.contentWindow.location.href, urlSansHash);
+  });
+  self.location.hash = "heya";
+}, "document.open() and document's URL containing a fragment");
```
Comment (domenic) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url-fragment.window.js]: Add "entry is not relevant" label

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url-fragment.window.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+async_test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"))
+  t.add_cleanup(() => { frame.remove(); });
+  frame.src = "resources/url-frame.html#heya";
+  window.testDone = t.step_func_done((beforeURL, afterURL) => {
+    assert_equals(beforeURL, frame.src);
+    assert_equals(afterURL, frame.src);
+    assert_equals(frame.contentDocument.URL, frame.src);
+  });
+}, "document.open() and document's URL containing a fragment (entry is current)");
```
Comment (domenic) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url-fragment.window.js]: s/current/relevant as that's what the spec tests

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url-fragment.window.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+window.testDone = undefined;
+async_test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"))
+  t.add_cleanup(() => { frame.remove(); });
+  frame.src = "resources/url-frame.html#heya";
+  window.testDone = t.step_func_done((beforeURL, afterURL) => {
+    assert_equals(beforeURL, frame.src);
+    assert_equals(afterURL, frame.src);
+    assert_equals(frame.contentDocument.URL, frame.src);
```
Comment (domenic) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url-fragment.window.js]: Test location.href in this test too?

**Review by domenic — APPROVED** [review]
LGTM with suggestions

**Review by domenic — APPROVED** [review]

**Pull request merged**